test/nextjs.test.js

```diff
@@ -606,6 +607,217 @@ describe(`NextJS Middleware Test: ${printPath("[test/nextjs.test.js]")}`, functi
+        const handleCall = getAppDirRequestHandler(NextResponse);
+
+        const userEmail = `john.doe.${Date.now()}@supertokens.com`;
+        const requestInfo = {
+            method: "POST",
+            headers: { rid: "emailpassword" },
+            body: JSON.stringify({
+                formFields: [
+                    { id: "email", value: userEmail },
+                    { id: "password", value: "P@sSW0rd" },
+                ],
+            }),
+        };
+
+        const signUpRequest = new NextRequest("http://localhost:3000/api/auth/signup", requestInfo);
+
+        const signUpRes = await handleCall(signUpRequest);
+        assert.deepStrictEqual(signUpRes.status, 200);
+        const tokens = getSessionTokensFromResponse(signUpRes);
+        assert.notEqual(tokens.access, undefined);
+        assert.notEqual(tokens.refresh, undefined);
+
+        const unAuthenticatedRequest = new NextRequest("http://localhost:3000/api/get-user");
+
+        let sessionContainer = await getSSRSession(
+            unAuthenticatedRequest.cookies.getAll(),
+            unAuthenticatedRequest.headers
+        );
+
+        assert.equal(sessionContainer.hasToken, false);
+        assert.equal(sessionContainer.session, undefined);
+
+        const authenticatedRequest = new NextRequest("http://localhost:3000/api/get-user", {
+            headers: {
+                Authorization: `Bearer ${tokens.access}`,
+            },
+        });
+
+        sessionContainer = await getSSRSession(authenticatedRequest.cookies.getAll(), authenticatedRequest.headers);
+
+        assert.equal(sessionContainer.hasToken, true);
+        assert.equal(sessionContainer.session.getUserId(), process.env.user);
+    });
+
+    it("withSession", async function () {
+        const handleCall = getAppDirRequestHandler(NextResponse);
+
+        const userEmail = `john.doe.${Date.now()}@supertokens.com`;
+        const requestInfo = {
+            method: "POST",
+            headers: { rid: "emailpassword" },
+            body: JSON.stringify({
+                formFields: [
+                    { id: "email", value: userEmail },
+                    { id: "password", value: "P@sSW0rd" },
+                ],
+            }),
+        };
+
+        const signUpRequest = new NextRequest("http://localhost:3000/api/auth/signup", requestInfo);
+
+        const signUpRes = await handleCall(signUpRequest);
+        assert.deepStrictEqual(signUpRes.status, 200);
+        const tokens = getSessionTokensFromResponse(signUpRes);
+        assert.notEqual(tokens.access, undefined);
+        assert.notEqual(tokens.refresh, undefined);
+
+        const unAuthenticatedRequest = new NextRequest("http://localhost:3000/api/get-user");
+
+        const unAuthenticatedResponse = await withSession(unAuthenticatedRequest, async (session) => {
+            if (!session) {
+                return new NextResponse("Authentication required", { status: 401 });
+            }
```

Comment:
this if statement will never be true cause we are not doing session verification optionally here. So remove this

Reply:
done